lib/local-links-manager/import/csv_downloader.rb

```diff
@@ -5,30 +5,50 @@ class Error < RuntimeError; end
   class DownloadError < Error; end
   class MalformedCSVError < Error; end
 
-  def initialize(csv_url, header_conversions = {})
+  def initialize(csv_url, header_conversions: {}, encoding: 'windows-1252')
     @csv_url = csv_url
     @header_conversions = header_conversions
+    @encoding = encoding
   end
 
-  def download
-    CSV.parse(downloaded_csv,
-              headers: true,
-              header_converters: field_name_converter)
+  def each_row(&block)
+    download do |csv|
+      csv.each(&block)
+    end
+  end
 
+  def download(&block)
+    downloaded_csv do |data|
+      block.call(
+        CSV.parse(
+          data,
+          headers: true,
+          header_converters: field_name_converter
+        )
+      )
+    end
   rescue CSV::MalformedCSVError => e
     raise MalformedCSVError, "Error #{e.class} parsing CSV in #{self.class}"
   end
 
 private
 
-  def downloaded_csv
-    response = Net::HTTP.get_response(URI.parse(@csv_url))
+  def downloaded_csv(&block)
+    Tempfile.create(['local_links_manager_import', @csv_url.gsub(/[^0-9A-z.\-]+/, '_'), 'csv']) do |temp_file|
+      temp_file.set_encoding('ascii-8bit')
```

Comment:
My guess would be that because there are no invalid byte sequences in ascii-8bit - so we can write any bytes we'd like without getting an encoding error.  Then we can later when we're ready to read the data out (L49 below) we set it to convert from the encoding we think it is into utf8 as we read.  TBH: I'm not sure why we don't set it to the encoding we think it is straight off - but I assume there's some valid reason.  I'll see if the git logs for publisher shed any light on this approach.